### ISSUE TITLE
Advantage and disadvantage logic fails on certain cases.

### ISSUE THREAD

**Issue by pmfirestone** [sbergot/mothership-assistant#6]
An example on page 20 of the 1e PSG fails: 

> My worst Save is Sanity at 14. So I get a 22 and a 56. So that would be a Crit Fail, but with Advantage it’s just a normal fail.

Expected behavior (in this case): normal fail.
Actual behavior (in this case): critical fail.

Advantage and disadvantage do not equal higher and lower rolls: there are cases where the "worse" result is the lower one, or the "better" result the higher one. Consider if both rolls succeed but the higher is a critical. When rolling advantage, the result should be a critical success, but in the current implementation rolling advantage in this case will report a normal success.

**Comment by pmfirestone** [comment]
I've fixed this on my fork, but I did it by way of a major refactor of the types: StatRoll and SaveRoll are now a single type, Roll, and the change is propagated through the whole code base. Does that seem like a reasonable solution? I'll prepare a clean pull request for that fix this weekend.

**Comment by sbergot** [comment]
Yes it makes sense. I don't remember why I split the two to be honest.
